test/fixtures.py

```diff
@@ -239,20 +269,30 @@ def __init__(self, host, port, broker_id, zk_host, zk_port, zk_chroot,
         self.child = None
         self.running = False
 
+        self._client = None
+
+    def bootstrap_server(self):
+        return '%s:%d' % (self.host, self.port)
+
     def kafka_run_class_env(self):
         env = super(KafkaFixture, self).kafka_run_class_env()
         env['LOG_DIR'] = os.path.join(self.tmp_dir, 'logs')
         return env
 
     def out(self, message):
-        log.info("*** Kafka [%s:%s]: %s", self.host, self.port or '(auto)', message)
+        log.info("*** Kafka [%s:%s]: %s", self.host, self.port or '(auto)', message.decode('utf-8') if isinstance(message, bytes) else message)
 
     def open(self):
         if self.running:
             self.out("Instance already running")
             return
 
-        self.tmp_dir = tempfile.mkdtemp()
+        # Create directories
+        if self.tmp_dir is None:
+            self.tmp_dir = tempfile.mkdtemp()
+            os.mkdir(os.path.join(self.tmp_dir, "logs"))
+            os.mkdir(os.path.join(self.tmp_dir, "data"))
```

Comment:
IIRC pytest has a built-in tmp dir fixture that can be used... is that relevant here at all? I'm not sure how these directories are actually used...

Reply:
Yes, it's relevant. I had a look at the pytest tmpdir fixture and will change the code to use it instead